subsys/bluetooth/audio/cap_commander.c

```diff
@@ -399,14 +400,302 @@ int bt_cap_commander_broadcast_reception_start(
 
 	return 0;
 }
-#endif /* CONFIG_BT_BAP_BROADCAST_ASSISTANT */
+
+static void
+copy_broadcast_reception_stop_param(struct bt_bap_broadcast_assistant_mod_src_param *mod_src_param,
+				    struct cap_broadcast_reception_stop *stop_param)
+{
+	mod_src_param->src_id = stop_param->src_id;
+	mod_src_param->pa_sync = false;
+	mod_src_param->pa_interval = BT_BAP_PA_INTERVAL_UNKNOWN;
+	mod_src_param->num_subgroups = stop_param->num_subgroups;
+
+	mod_src_param->subgroups = stop_param->subgroups;
+}
+
+static void cap_commander_broadcast_assistant_recv_state_cb(
+	struct bt_conn *conn, int err, const struct bt_bap_scan_delegator_recv_state *state)
+{
+	struct bt_cap_common_proc *active_proc = bt_cap_common_get_active_proc();
+	struct bt_conn *cap_conn = NULL;
+	char le_addr[BT_ADDR_LE_STR_LEN];
+	uint16_t total_bis_sync;
+	uint8_t src_id = 0;
+
+	if (err != 0) {
+		LOG_DBG("BASS recv state read failed (%d)\n", err);
+		return;
+	}
+
+	if (state == NULL) {
+		/* Empty receive state */
+		return;
+	}
+
+	bt_addr_le_to_str(&state->addr, le_addr, sizeof(le_addr));
+
+	LOG_DBG("BASS recv state: src_id %u, addr %s, sid %u, sync_state %u", state->src_id,
+		le_addr, state->adv_sid, state->pa_sync_state);
+
+	if (state->encrypt_state == BT_BAP_BIG_ENC_STATE_BAD_CODE) {
+		LOG_DBG("Encryption state is BT_BAP_BIG_ENC_STATE_BAD_CODE");
+		return;
+	}
+
+	if (bt_cap_common_conn_in_active_proc(conn) &&
+	    active_proc->proc_type == BT_CAP_COMMON_PROC_TYPE_BROADCAST_RECEPTION_STOP) {
+		total_bis_sync = 0;
+		for (uint8_t i = 0; i < state->num_subgroups; i++) {
+			const struct bt_bap_bass_subgroup *subgroup = &state->subgroups[i];
+
+			total_bis_sync |= subgroup->bis_sync;
+		}
+		for (size_t i = 0; i < active_proc->proc_cnt; i++) {
+			if (state->src_id == active_proc->proc_param.commander[i]
+						     .broadcast_reception_stop.src_id &&
+			    conn == active_proc->proc_param.commander[i].conn) {
+				src_id = state->src_id;
+				cap_conn = conn;
+			}
+		}
```

Comment:
What's the purpose of this loop?

Won't `src_id` always be `state->src_id` and `cap_conn` always be `conn`?